src/java.base/share/classes/sun/security/ssl/SSLSocketOutputRecord.java

```diff
@@ -169,7 +169,8 @@ void encodeHandshake(byte[] source,
             for (int limit = (offset + length); offset < limit;) {
 
                 int remains = (limit - offset) + (count - position);
```

Comment:
```suggestion
                int remains = (limit - offset);
```

Reply:
I tried this and it caused a lockup in one of the tests.  I see why your think this is the right change, but it isn't proving out in the testing

Reply:
That's very interesting! Which test was it? Was it with or without #19465?

Reply:
I do run the tests layered on the current #19465, even though this PR is separate.